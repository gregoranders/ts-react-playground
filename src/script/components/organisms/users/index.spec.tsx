import React, { RecoilRoot, act, create, mount } from '@app/testUtils';

import * as TestSubject from '@organisms/users';

describe(`${TestSubject.UsersModelView.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  it('snapshot', (): void => {
    const snapshot = create(
      <RecoilRoot>
        <TestComponent />
      </RecoilRoot>,
    );
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('mount', () => {
    const testSubject = mount(
      <RecoilRoot>
        <TestComponent />
      </RecoilRoot>,
    );
    expect(testSubject.text()).toBe('Generate Users [0]');
  });

  it('mount', () => {
    const testSubject = mount(
      <RecoilRoot>
        <TestComponent />
      </RecoilRoot>,
    );
    act(() => {
      testSubject.find('button').simulate('click');
    });
    expect(testSubject.text()).toContain('Generate Users [15]');
  });
});
