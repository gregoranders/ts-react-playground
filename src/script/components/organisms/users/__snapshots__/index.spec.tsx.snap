// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersModelView snapshot 1`] = `
Array [
  <button
    disabled={false}
    onClick={[Function]}
  >
    Generate Users
  </button>,
  <table>
    <tbody>
      <tr>
        <td />
      </tr>
    </tbody>
  </table>,
]
`;
